src/data/fintype/basic.lean

```diff
@@ -1923,6 +1925,22 @@ begin
     refine of_equiv e (h_option ih), },
 end
 
+/-- An induction principle for finite types, analogous to `nat.rec`. It effectively says
+that every `fintype` is either `empty` or `option α`, up to an `equiv`. -/
+@[elab_as_eliminator]
+lemma induction_empty_option' {P : Π (α : Type u) [fintype α], Prop}
+  (of_equiv : ∀ α β [fintype β] (e : α ≃ β), @P α (@fintype.of_equiv α β ‹_› e.symm) → @P β ‹_›)
```

Comment:
Is this spelling better or worse?
```suggestion
  (of_equiv : ∀ α β [fintype α] [fintype β] (e : α ≃ β), by exactI P α → P β)
```

Reply:
I prefer to assume that the instance on `α` comes from `e` because this way we have better defeq.